web/src/views/FileBrowserView/PublishFileBrowser.vue

```diff
@@ -188,7 +187,28 @@ export default {
         this.owners = (await publishRest.owners(identifier)).data
           .map((x) => x.username);
 
-        let mapped = data.map((x) => ({ name: x, folder: isFolder(x) }));
+        // flatten the assetPaths object into an array:
+        let mapped = [];
+
+        if (data.files) {
+          mapped = Object.keys(data.files).map(
+            (name) => ({ name, folder: false, ...data.files[name] }),
+          );
+          // set download links and uuids (needed for deleting) for files
+          mapped.forEach((item) => {
+            this.$set(this.itemData, item.name, {
+              download: item.download_url,
+              uuid: item.asset_id,
+            });
+          });
+        }
+
+        if (data.folders) {
+          mapped = mapped.concat(Object.keys(data.folders).map(
+            (name) => ({ name: `${name}/`, folder: true, ...data.folders[name] }),
+          ));
+        }
+
```

Comment:
Based on the above comments, we might be able to simplify this logic like so:

```typescript
        return [
          ...location !== rootDirectory ? [{ name: parentDirectory, folder: true }] : [],
          ...Object.keys(data.files).map(
            (key) => ({ ...data.files[key], name: key, folder: false }),
          ),
          ...Object.keys(data.folders).map(
            (key) => ({ ...data.folders[key], name: `${key}/`, folder: true }),
          ),
        ];
```

Reply:
And thinking about this now, we should probably put folders first, so they're visible

Reply:
https://github.com/dandi/dandiarchive/pull/773/commits/17b25cc61215faaf90c56b9a73e4c1d1904ec4d7